App.js

```diff
@@ -1,52 +1,70 @@
 import 'react-native-gesture-handler';
 import _ from 'underscore';
 import * as React from 'react';
-import {NavigationContainer, DefaultTheme, CommonActions, getStateFromPath, useFocusEffect} from '@react-navigation/native';
-import {Text, View, Image, Pressable, Linking, Platform, BackHandler, Dimensions} from 'react-native';
+import {NavigationContainer, DefaultTheme,  getStateFromPath, useFocusEffect,  createNavigationContainerRef } from '@react-navigation/native';
+import {Text, View, Image, Pressable,  BackHandler, Dimensions, Platform}  from 'react-native';
 import {createNativeStackNavigator} from '@react-navigation/native-stack';
 import Constants from 'expo-constants';
 import createWebNavigator from './createWebNavigator';
+import * as Linking from 'expo-linking';
 
 const chevronStyle = {width: 30, height: 30, resizeMode: 'contain', marginRight: 10};
 const chatTitleStyle = {fontSize: 18, fontWeight: 'bold'};
 const titleStyle = {fontSize: 24, fontWeight: 'bold', flex: 1};
 
+const keysBlocklist = ['key', 'stale', 'routeNames']
+
+const stripNavigationState = (state) => {
+  const rawState = Array.isArray(state) ? [] : {};
+  for (const key of Object.keys(state)) {
+    if (keysBlocklist.includes(key)) {
+      continue
+    }
+    rawState[key] = typeof state[key] === 'object' ? stripNavigationState(state[key]) : state[key]
+  }
+  return rawState 
+}
+
 const config = {
-  initialRouteName: '',
+  initialRouteName: 'LeftHandNav',
   screens: {
     Chat: 'r/:id?',
     LeftHandNav: '',
     SettingsStack: {
+      initialRouteName: 'Settings',
+      path: 'settings',
       screens: {
-        Settings: 'settings',
-        About: 'settings/about',
+        Settings: '',
+        About: 'about',
       },
     },
     Search: 'search'
   },
 };
 
-const linking = {
-  prefixes: ['http://localhost', 'https://navigation-test-lime.vercel.app'],
-  config,
-  getStateFromPath(path, cfg) {
-    const state = getStateFromPath(path, cfg);
-    console.log('Get state from path: ', state);
 
-    // We need to add the LHN if it does not exist
-    if (!_.find(state.routes, r => r.name === 'LeftHandNav')) {
-      state.routes.splice(0, 0, {name: 'LeftHandNav'});
+const addChatRouteIfNecessary = (state) => {
```

Comment:
Can we update this to 

```suggestion
const fixState = (state) => {
```